### ISSUE TITLE
Stream.update does not honour the seed argument for the whole call

### ISSUE THREAD

**Issue by sachncs** [sachncs/laker#30]
## Problem

`Stream.update` does not re-seed the `Backend` or the local numpy RNG when a `seed` argument is provided. The `seed` is forwarded only to `CCCP.build` (for the random probes); the train/val split that happens later inside `search` / `bayes` is seeded separately, but the `update` path itself is not. The result is that running `Laker.update(x_new, y_new, seed=42)` twice on the same model does not yield the same kernel/prec because the model.embed reshape / cat order is non-deterministic on GPU.

## Evidence

`laker/stream.py:79-87`:

```python
with torch.no_grad():
    prec = self.core.build_prec(
        kernel.matvec,
        model.embed.shape[0],
        diag=kernel.diag(),
        seed=seed,
    )
    model.prec = prec
    model.coef, model.iters = self.core.solve(kernel, prec, y_ext, x0=x0)
```

The `seed` is used only in `build_prec`. The `kernel` itself is constructed from `model.embed` at `laker/stream.py:68`, which depends on the previous run's RNG state. For the *first* `update` after `fit`, the state is whatever the user left it in; for subsequent `update` calls, it depends on the previous `update`'s CCCP build (which consumed the generator).

`tests/test_stream.py` uses `torch.manual_seed(0)` *before* `Laker.fit` but never re-seeds before `Laker.update`, so the test is not exercising a reproducible streaming scenario.

## Root Cause

The `seed` parameter is plumbed but its semantic is inconsistent: in `Laker.fit` it controls the random probes only (not the data iteration), and in `Stream.update` it controls only the new probes for the rebuilt preconditioner. From a user's perspective, `seed=N` should make the whole call reproducible.

## Impact

Two `update` calls with the same `seed` and same `x_new`/`y_new` produce different kernel/prec because of accumulated generator state. This is the same kind of contract violation as the bayes-reproducibility bug, and the fix should follow the same pattern (seed the global torch/numpy RNG at the start of the call).

## Recommendation

At the top of `Stream.update`, if `seed is not None`:

```python
torch.manual_seed(int(seed))
np.random.seed(int(seed))
Backend.seed(int(seed))   # also sets LAKER_SEED env var
```

This is the same pattern that `Backend.load` and `Math.seed_set` already implement.

## Acceptance Criteria

- [ ] `Laker.update(x_new, y_new, seed=42)` called twice on a freshly-fit model produces identical `model.coef` and `model.prec.iso` values.
- [ ] A new test in `tests/test_stream.py` covers this property.
- [ ] `Stream.update` does not silently consume the global RNG state in a way that affects the user's other code.

## Verification

```python
m1 = Laker(embed_dim=4); m1.fit(x, y)
m2 = Laker(embed_dim=4); m2.fit(x, y)
m1.update(x_new, y_new, seed=42)
m2.update(x_new, y_new, seed=42)
assert torch.allclose(m1.coef, m2.coef)
assert m1.prec.iso == m2.prec.iso
```

## Dependencies

None.

## Risk

Low. Reproducibility is a contract improvement; no behaviour change for users who do not pass `seed`.

**Comment by sachncs** [comment]
Resolved in PR #32.
